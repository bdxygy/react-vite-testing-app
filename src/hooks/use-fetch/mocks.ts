export const initialMock = {
  data: {
    name: "Samsudin",
    age: 30,
    email: "samsudin@example.com",
  },
  error: false,
  isLoading: false,
};

export const secondMock = {
  data: {
    name: "Sumarsih",
    age: 25,
    email: "sumarsih@example.com",
  },
  error: false,
  isLoading: false,
};

export const errorMock = {
  data: {},
  error: true,
  isLoading: false,
};

export const loadingMock = {
  data: {},
  error: false,
  isLoading: true,
};
