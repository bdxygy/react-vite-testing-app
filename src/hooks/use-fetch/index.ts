import { initialMock } from "./mocks";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const useFetch = (): Record<string, any> => {
  return initialMock;
};
