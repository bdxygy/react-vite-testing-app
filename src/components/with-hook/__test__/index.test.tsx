import { render, screen } from "@testing-library/react";
import { WithHook } from "..";
import { useFetch } from "../../../hooks/use-fetch";
import {
  initialMock,
  errorMock,
  loadingMock,
  secondMock,
} from "../../../hooks/use-fetch/mocks";

jest.mock("../../../hooks/use-fetch");

const useFetchMock = useFetch as jest.MockedFunction<typeof useFetch>;

useFetchMock.mockReturnValue(initialMock);

describe("with-hook", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should component render", () => {
    const component = render(<WithHook />);

    const h1 = screen.getByText("Samsudin");
    expect(h1).toBeInTheDocument();

    expect(component.container).toMatchSnapshot();
  });

  it("should component render with data Tester", () => {
    useFetchMock.mockReturnValue(secondMock);

    const component = render(<WithHook />);

    const h1 = screen.getByText("Sumarsih");

    expect(h1).toBeInTheDocument();

    expect(component.container).toMatchSnapshot();
  });

  it("should component render with error", () => {
    useFetchMock.mockReturnValue(errorMock);

    const component = render(<WithHook />);

    const h1 = screen.getByText("Error");
    expect(h1).toBeInTheDocument();

    expect(component.container).toMatchSnapshot();
  });

  it("should component render with loading", () => {
    useFetchMock.mockReturnValue(loadingMock);

    const component = render(<WithHook />);

    const h1 = screen.getByText("Loading...");
    expect(h1).toBeInTheDocument();

    expect(component.container).toMatchSnapshot();
  });
});
