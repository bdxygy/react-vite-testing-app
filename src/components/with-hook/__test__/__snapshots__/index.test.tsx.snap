// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-hook should component render 1`] = `
<div>
  <h1>
    Samsudin
  </h1>
</div>
`;

exports[`with-hook should component render with data Tester 1`] = `
<div>
  <h1>
    Sumarsih
  </h1>
</div>
`;

exports[`with-hook should component render with error 1`] = `
<div>
  <h1>
    Error
  </h1>
</div>
`;

exports[`with-hook should component render with loading 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;
